class/practice_6.1/message_generator.cpp

```diff
@@ -0,0 +1,42 @@
+#include "message_dispatcher.hpp" 
+
+void generateMessages(MessageDispatcher& dispatcher, const std::string& threadName) {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> counterDist(1, 10);
+    std::uniform_real_distribution<> doubleDist(0.0, 10.0);
+
+    while (true) {
+        int messageType = gen() % 3;
+
+        switch (messageType) {
+            case 0: {
+                std::string text = "Green message: ";
+                text += std::to_string(counterDist(gen));
+                GreenMessage greenMessage{text, counterDist(gen)};
+                std::cout << threadName << " published GreenMessage: " << greenMessage.text << " " << greenMessage.counter << std::endl;
```

Comment:
std::println can make it more readable and we will not need extra reallocations on string concatenations and no explicit to_strng call